lib/models/personal_profile.dart

```diff
@@ -72,13 +81,15 @@ class SupporterMetadata {
   factory SupporterMetadata.fromMap(Map map) {
     final type = map["\$type"];
     return switch (type) {
-      "patreon" => PatreonSupporter.fromMap(map),
-      _ => SupporterMetadata(),
+      "patreon" => SubscriptionSupporter.fromMap(map),
+      "stripe" => SubscriptionSupporter.fromMap(map),
```

Comment:
The first two cases can be combined to `"patreon" || "stripe" => SubscriptionSupporter.fromMap(map),`